syntax = "proto2";

package HDDS_API;

enum ErrCode {
	FAILURE_INVALID_MESSAGE = 1000;
	FAILURE_INVALID_KEY     = 1001;
	FAILURE_INVALID_VALUE   = 1002;
	FAILURE_SET_UNSUPPORTED = 1003;
	FAILURE_GET_FAILED      = 1004;
	FAILURE_SET_FAILED      = 1005;
}

message Property {
    required string key     = 1;
    required string value   = 2;
}

message ErrorMessage {
    required uint32 error_code          = 1;
    required string error_description   = 2;
}

message ValueResp {
    required bool success  = 1;
    required Property keyValue = 2;
    optional ErrorMessage error = 3;
}


message GetReq {
	repeated string key = 1;
}

message GetResp {
	repeated ValueResp values = 1;
	required bool success = 2;
	optional ErrorMessage error = 3;
}

message SetReq {
	repeated Property values = 1;
}

message SetResp {
	repeated ValueResp values = 1;
	required bool success = 2;
	optional ErrorMessage error = 3;
}